really-core/src/main/scala/io/really/gorilla/SubscribeAggregator.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2014-2015 Really Inc. <http://really.io>
+ */
+
+package io.really.gorilla
+
+import akka.actor.{ ActorLogging, ActorRef, Actor }
+import akka.contrib.pattern.Aggregator
+import io.really.R
+import io.really.RequestContext
+import io.really.WrappedSubscriptionRequest.WrappedSubscribe
+import io.really.gorilla.SubscriptionManager.{ SubscriptionDone, SubscribeOnR }
+import io.really.protocol.{ SubscriptionFailure, SubscriptionBody }
+import scala.collection.mutable.ArrayBuffer
+import io.really.ReallyGlobals
+
+class SubscribeAggregator(subscriptionManager: ActorRef, globals: ReallyGlobals) extends Actor with Aggregator with ActorLogging {
+
+  import context._
+  import SubscribeAggregator._
+
+  expectOnce {
+    case WrappedSubscribe(subscribeRequest, pushChannel) =>
+      new SubscribeAggregatorImpl(subscribeRequest.ctx, sender(), subscribeRequest.body, pushChannel)
+    case msg =>
+      log.error(s"Subscribe Aggregator got an unexpected response: $msg and going to die")
+      sender() ! UnsupportedResponse
+      context.stop(self)
+  }
+
+  class SubscribeAggregatorImpl(ctx: RequestContext, requestDelegate: ActorRef, body: SubscriptionBody,
+      pushChannel: ActorRef) {
+
+    val results = ArrayBuffer.empty[R]
+    var processedCount = 0
+
+    if (body.subscriptions.size > 0) {
```

Comment:
I feel it is the right decision, and this was discussed with @AhmedSoliman before implementation, the point is that the `SubscriptionManager` is a singleton per bucket, so I feel it is good to dispatch this task to another actor who will die after finishing the task...Making the `SubscriptionManager` aggregates the subscriptions will IMHO add more complexity